# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/xalan-c/hotfix-template.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Fix the unused writeSafe() template referencing non-existing members,
rejected by GCC 15+ -Wtemplate-body.

diff -Nrup xalan_c-1.12.orig/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp xalan_c-1.12/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp
--- xalan_c-1.12.orig/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp	2020-06-07 11:17:25.000000000 +0200
+++ xalan_c-1.12/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp	2026-09-27 18:32:51.412827044 +0200
@@ -320,13 +320,13 @@ public:
                 {
                     XalanUnicodeChar    value = decodeUTF16SurrogatePair(ch, theChars[i+1],  getMemoryManager());
 
-                    if (this->m_isPresentable(value))
+                    if (m_predicate(value))
                     {
                         write(value);
                     }
                     else
                     {
-                        this->writeNumberedEntityReference(value);
+                        writeNumericCharacterReference(value);
                     }
 
                     ++i;
